src/parser/parser_antlr.cpp

```diff
@@ -0,0 +1,271 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *   Andrew Reynolds, Mathias Preiner, Gereon Kremer
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2022 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * Parser state implementation.
+ */
+
+#include "parser/parser_antlr.h"
+
+#include <clocale>
+#include <fstream>
+#include <iostream>
+#include <iterator>
+#include <sstream>
+#include <unordered_set>
+
+#include "api/cpp/cvc5.h"
+#include "base/check.h"
+#include "base/output.h"
+#include "expr/kind.h"
+#include "parser/api/cpp/command.h"
+#include "parser/input.h"
+#include "parser/parser_exception.h"
+#include "parser/smt2/smt2_input.h"
+
+using namespace std;
+
+namespace cvc5 {
+namespace parser {
+
+class Parser::IncludeFileCache
+{
+ public:
+  IncludeFileCache() {}
+  ~IncludeFileCache()
+  {
+    for (size_t i = 0, isize = d_inCreated.size(); i < isize; i++)
+    {
+      d_inCreated[i]->free(d_inCreated[i]);
+    }
+  }
+  std::vector<pANTLR3_INPUT_STREAM> d_inCreated;
+};
+
+Parser::Parser() : d_done(true), d_canIncludeFile(true) {}
+
+Parser::~Parser()
+{
+  for (std::list<Command*>::iterator iter = d_commandQueue.begin();
+       iter != d_commandQueue.end();
+       ++iter)
+  {
+    Command* command = *iter;
```

Comment:
Is there any reason why these are not std::unique_ptr? If we'll use this code with the new parser, we should use unique_ptr.

Reply:
This won't be used in the new parser.   The new parser uses unique_ptr up to the point where the 2 parsers have to share an interface.